src/gmail.js

```diff
@@ -1505,23 +1510,68 @@ var Gmail = function(localJQuery) {
     return selected_emails;
   }
 
+  /*
+   * @case   https://mail.google.com/mail/u/0/#inbox
+   * @case   https://mail.google.com/mail/u/0/#inbox?compose=14c0f011dffd86bc
+   * @case   https://mail.google.com/mail/u/0/#inbox/p2
+   * @case   https://mail.google.com/mail/u/0/#inbox/p2?compose=14c0f011dffd86bc
+   * @case   https://mail.google.com/mail/u/0/#label/name
+   * @case   https://mail.google.com/mail/u/0/#label/name?compose=14c0f011dffd86bc
+   * @case   https://mail.google.com/mail/u/0/#label/name/p2
+   * @case   https://mail.google.com/mail/u/0/#label/name/p2?compose=14c0f011dffd86bc
+   */
   api.get.current_page = function() {
-    var hash  = window.location.hash.split('#').pop().split('?').shift();
-    var pages = ['sent', 'inbox', 'starred', 'drafts', 'imp', 'chats', 'all', 'spam', 'trash', 'settings'];
-
-    var page = null;
-
-    if($.inArray(hash, pages) > -1) {
-      page = hash;
+    var pages = ['sent', 'inbox', 'starred', 'drafts', 'imp', 'chats', 'all', 'spam', 'trash'];
```

Comment:
Since `settings` & `settings/*` can't contain any threads (eg. `settings/1748vsbdnd`), it's returned through the rest of the conditions/logic in the function. This function is only used to filter the regular expression to pages that could possible have a thread as part of the hash.